frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/provision-source.tsx

```diff
@@ -69,19 +75,33 @@ export const ProvisionSourceComponent: React.FC<ProvisionSourceComponentProps> =
     const provisionSourceValue = iGetFieldValue(provisionSourceField);
     const sources = iGet(provisionSourceField, 'sources');
 
+    const [isSourceOpen, setIsSourceOpen] = React.useState(false);
+
+    const onSelect = (e, v) => {
+      onChange(VMSettingsField.PROVISION_SOURCE_TYPE, v.toString());
+      setIsSourceOpen(false);
+    };
+
     return (
       <FormFieldRow field={provisionSourceField} fieldType={FormFieldType.SELECT}>
-        <FormField>
-          <FormSelect onChange={(v) => onChange(VMSettingsField.PROVISION_SOURCE_TYPE, v)}>
-            <FormSelectPlaceholderOption
-              placeholder={getPlaceholder(VMSettingsField.PROVISION_SOURCE_TYPE)}
-              isDisabled={!!provisionSourceValue}
+        <Select
```

Comment:
can you please create a new type for FormField and use it here? This will allow to pass correct values and metadata from redux to this component.